### PR TITLE
Add Mapbox GL JS typings

### LINKED ISSUE #7774: Add definition for mapbox-gl
There is only a def for mapbox.js, one for mapbox-gl.js would be great!

### PR DESCRIPTION
**Pull request by dobrud**
case 1. Add a new type definition.
- [X] checked compilation succeeds with `--target es6` and `--noImplicitAny` options.
- [X] has correct [naming convention](http://definitelytyped.org/guides/contributing.html#naming-the-file)
- [x] has a [test file](http://definitelytyped.org/guides/contributing.html#tests) with the suffix of  `-tests.ts` or `-tests.tsx`.

fixes #7774

### CODE REVIEW

File: mapbox-gl/mapbox-gl-tests.ts [hunk 1]
```diff
@@ -0,0 +1,376 @@
+// These examples are from Mapbox (https://www.mapbox.com/mapbox-gl-js/examples/) and therefore under the copyright of Mapbox Inc.
+/// <reference path="./mapbox-gl.d.ts" />
+
+import * as mapboxgl from './mapbox-gl';
```
Comment (vvakame) [inline mapbox-gl/mapbox-gl-tests.ts]: use `'mapbox-gl'` instead of `'./mapbox-gl'`.

File: mapbox-gl/mapbox-gl.d.ts [hunk 2]
```diff
@@ -0,0 +1,469 @@
...
+		isPaintValueFeatureConstant(name: any): boolean; // Todo: not yet specified in documentation
+		isHidden(zoom: any): boolean; // Todo: not yet specified in documentation
+		updatePaintTransitions(classes: any, options: any, globalOptions: any, animationLoop: any): void; // Todo: not yet specified in documentation
+		updatePaintTransition(name: any, classes: any, options: any, globalOptions: any, animationLoop: any): void; // Todo: not yet specified in documentation
+		recalculate(zoom: any, zoomHistory: any): void; // Todo: not yet specified in documentation
+		serialize(options: any): Object; // Todo: not yet specified in documentation
+	}
+}
+
+export = mapboxgl;
```
Comment (vvakame) [inline mapbox-gl/mapbox-gl.d.ts]:
```
declare module 'mapbox-gl' {
    export = mapboxgl;
}
```

**Comment by filoxo** [comment]
@dobrud Could you provide an explanation as to why this was closed without merging? Would be great to make these types available.

**Comment by dobrud** [comment]
We remained with the choice to locate the type definition in our own repository which you can find [here](https://github.com/dobrud/typed-mapbox-gl).

Please refer also to [this discussion](https://github.com/mapbox/mapbox-gl-js/issues/2440) on the main project repo.

**Pull request closed without merging**